src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -1620,7 +1620,12 @@ impl<T: AstInfo> AstDisplay for CreateSecretStatement<T> {
         }
         f.write_node(&self.name);
         f.write_str(" AS ");
-        f.write_node(&self.value);
+
+        if f.redacted() {
```

Comment:
I don't think this is necessary! values already redact themselves.

Reply:
Right! But secrets can also be derived from expressions, and when storing the `create_sql` for a `SECRET` we [replace the entire expression with *******](https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/plan/statement/ddl.rs#L3735), so I figured we could follow a similar pattern here and replace the entire expression with `'<REDACTED>'`. More than happy to get rid of this extra bit of logic though and let the Value redacting kick in!

Reply:
Ah! Ok, I'm convinced by the argument that the expression should also be redacted.